### PR TITLE
refactor(views/*): rewrite in TypeScript

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: @types/ucontent.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+declare module "ucontent" {
+  type TemplateFunction<T> = (
+    template: TemplateStringsArray,
+    ...values: unknown[]
+  ) => T;
+
+  class UContent extends String {
```
Comment (marcoscaceres) [inline @types/ucontent.d.ts:7]: Q: why "UContent"? It's not immediately obvious what that means. Maybe just needs a comment.

**Review by marcoscaceres — APPROVED** [review]
Just a small question... otherwise looks fine.

File: @types/ucontent.d.ts (hunk 1 above)
Comment (sidvishnoi) [reply @types/ucontent.d.ts:7]: Umm.. because we're defining types for `uncontent` package?

Comment (marcoscaceres) [reply @types/ucontent.d.ts:7]: Ok, but what *is* `UContent`... in particular, what's the "U" standard for?

Comment (sidvishnoi) [reply @types/ucontent.d.ts:7]: U is short for micro (mu). It's the package name. Am I missing something?

Comment (marcoscaceres) [reply @types/ucontent.d.ts:7]: Oh, I see now... yeah, somehow I missed the whole "mu" package inclusion. Seems I forgot we were using that. Just seems odd that we would be defining types of external packages, which is also why I got confused.

Comment (sidvishnoi) [reply @types/ucontent.d.ts]: Don't need this file with https://github.com/marcoscaceres/respec.org/pull/176/commits/7947b2e949ff60d2a9d64edcc23f116956143dda :tada:

**Review by marcoscaceres — APPROVED** [review]

File: @types/ucontent.d.ts (hunk 1 above)
Comment (marcoscaceres) [reply @types/ucontent.d.ts]: Unless comment turned out to be useful! :D

**Pull request merged**